test/e2e/app/app.go

```diff
@@ -356,21 +363,21 @@ func (app *Application) CheckTx(_ context.Context, req *abci.CheckTxRequest) (*a
 		return &abci.CheckTxResponse{Code: kvstore.CodeTypeOK, GasWanted: 1}, nil
 	}
 	lane := extractLane(value)
-	return &abci.CheckTxResponse{Code: kvstore.CodeTypeOK, GasWanted: 1, Lane: lane}, nil
+	return &abci.CheckTxResponse{Code: kvstore.CodeTypeOK, GasWanted: 1, LaneId: lane.GetId()}, nil
 }
 
 // extractLane returns the lane field if value is a Payload, otherwise returns 0.
-func extractLane(value string) uint32 {
+func extractLane(value string) payload.Lane {
```

Comment:
Return lane id and update doc-comment.

Reply:
I replaced it as as string because it is returned as such in checkTx and we'd have to convert it right back.